sidecar/config/flags.go

```diff
@@ -22,31 +22,36 @@ import (
 )
 
 const (
-	configFlag              = "config"
-	registerFlag            = "register"
-	proxyFlag               = "proxy"
-	serviceFlag             = "service"
-	endpointHostFlag        = "endpoint_host"
-	endpointPortFlag        = "endpoint_port"
-	endpointTypeFlag        = "endpoint_type"
-	registryBackendFlag     = "registry_backend"
-	registryURLFlag         = "registry_url"
-	registryTokenFlag       = "registry_token"
-	registryPollFlag        = "registry_poll"
-	kubernetesURLFlag       = "kubernetes_url"
-	kubernetesTokenFlag     = "kubernetes_token"
-	kubernetesNamespaceFlag = "kubernetes_namespace"
-	eurekaURLFlag           = "eureka_url"
-	controllerURLFlag       = "controller_url"
-	controllerTokenFlag     = "controller_token"
-	controllerPollFlag      = "controller_poll"
-	superviseFlag           = "supervise"
-	healthchecksFlag        = "healthchecks"
-	logLevelFlag            = "log_level"
-	dnsFlag                 = "dns"
-	dnsConfigPortFlag       = "dns_port"
-	dnsConfigDomainFlag     = "dns_domain"
-	debugFlag               = "debug"
+	configFlag                  = "config"
+	registerFlag                = "register"
+	proxyFlag                   = "proxy"
+	proxyTypeFlag               = "proxy_type"
```

Comment:
So, we have to consider the case that the proxy could be just doing TCP proxying and no HTTP. In which case, TLS still applies. Which is why I suggested TLS being a boolean attribute.